tests/test_transformer.py

```diff
@@ -295,6 +298,15 @@ def test_transformer__RestrictingNodeTransformer__guard_iter(compile, mocker):
     _getiter_.assert_called_once_with(it)
     _getiter_.reset_mock()
 
+    ret = glb['nested_generator']((0, 1, 2), (1, 2))
+    assert isinstance(ret, types.GeneratorType)
+    assert list(ret) == [2, 3, 3, 4]
+    _getiter_.assert_has_calls([
+        mocker.call((0, 1, 2)),
+        mocker.call((1, 2)),
+        mocker.call((1, 2))])
+    _getiter_.reset_mock()
```

Comment:
This method should split up into several ones. The setup an teardown could be done in a fixture so the test functions do not get bloated.